utils/check_copies.py

```diff
@@ -602,8 +604,23 @@ def check_codes_match(observed_code: str, theoretical_code: str) -> Optional[int
     _re_func_match = re.compile(r"def\s+([^\(]+)\(")
     for re_pattern in [_re_class_match, _re_func_match]:
         if re_pattern.match(observed_code_header) is not None:
-            observed_obj_name = re_pattern.search(observed_code_header).groups()[0]
-            theoretical_name = re_pattern.search(theoretical_code_header).groups()[0]
+            try:
+                observed_obj_name = re_pattern.search(observed_code_header).groups()[0]
+            except Exception:
+                raise ValueError(
+                    "Tried to split a class or function. It did not work. Error comes from: \n```\n"
```

Comment:
The error message is a bit confusing here. The `re_pattern` is used to extract the class or function name, not to split the blocks (unlike the above one)

Reply:
feel free to update it. 😄